src/linear_algebra/finsupp.lean

```diff
@@ -822,6 +822,46 @@ end prod
 
 end finsupp
 
+section fintype
+
+variables (α M R : Type*) [semiring R] [add_comm_monoid M] [module R M] (v : α → M) [fintype α]
```

Comment:
The `finsupp` files make a lot of variables explicit for, IMO, little reason. Could you try keeping them implicit?
```suggestion
variables {α M R : Type*} [semiring R] [add_comm_monoid M] [module R M] (v : α → M) [fintype α]
```